WordPress/src/main/java/org/wordpress/android/ui/qrcodeauth/QRCodeAuthFragment.kt

```diff
@@ -25,41 +35,71 @@ import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Content
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Error
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Loading
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Scanning
+import org.wordpress.android.ui.qrcodeauth.compose.state.ContentState
+import org.wordpress.android.ui.qrcodeauth.compose.state.ErrorState
+import org.wordpress.android.ui.qrcodeauth.compose.state.LoadingState
 import org.wordpress.android.ui.utils.UiHelpers
 import org.wordpress.android.viewmodel.observeEvent
 import javax.inject.Inject
 
 @AndroidEntryPoint
 @Suppress("TooManyFunctions")
-class QRCodeAuthFragment : Fragment(R.layout.qrcodeauth_fragment) {
+class QRCodeAuthFragment : Fragment() {
     @Inject lateinit var uiHelpers: UiHelpers
 
-    private val viewModel: QRCodeAuthViewModel by viewModels()
+    private val qrCodeAuthViewModel: QRCodeAuthViewModel by viewModels()
     private val dialogViewModel: BasicDialogViewModel by activityViewModels()
 
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View = ComposeView(requireContext()).apply {
+        setContent {
+            AppTheme {
+                QRCodeAuthScreen()
+            }
+        }
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        with(QrcodeauthFragmentBinding.bind(view)) {
-            initBackPressHandler()
-            observeViewModel()
-            initViewModel(savedInstanceState)
+        initBackPressHandler()
+        initViewModel(savedInstanceState)
+        observeViewModel()
+    }
+
+    @Composable
+    private fun QRCodeAuthScreen(viewModel: QRCodeAuthViewModel = viewModel()) {
+        VerticalScrollBox(
+                alignment = Alignment.CenterStart,
+                modifier = Modifier.fillMaxSize()
+        ) {
+            val uiState by viewModel.uiState.collectAsState()
+            uiState.run {
```

Comment:
Why do you do this? I think it should be enough to do: 
```
val uiState by viewModel.uiState.collectAsState()
when (uiState) {
...
}
```

Reply:
I was trying to do this while working on the PR, but got held back by this error we're getting:

<img width="993" alt="Screenshot 2022-07-27 at 12 58 13" src="https://user-images.githubusercontent.com/4588074/181231169-31d6ce2a-c1c3-48e9-aa79-2a40094a6887.png">

It will also block me if I'm trying to cast it manually:
<img width="776" alt="Screenshot 2022-07-27 at 12 59 48" src="https://user-images.githubusercontent.com/4588074/181231564-879b542a-fcb5-406c-9808-3006a2539d30.png">

Reply:
@ovitrif I think you should pass `uiState` to the composables, and not `viewModel`, something like this 🤔 :

```kotlin
        val uiState by viewModel.uiState.collectAsState()
        when (uiState) {
                is Content -> ContentState(uiState)
                is Error -> ErrorState(uiState)
                is Loading -> LoadingState()
                is Scanning -> Unit
            }
```

Reply:
> @ovitrif I think you should pass uiState to the composables, and not viewModel

Did I really do that 🤦 🤣 🤷 , I might have been out of the solar system 😅.

My bad, I was posting wrong screenshots, made in a hurry because I knew I've tried this but with the right code; and then the screenshots were with the wrong code 🤦 :

<img width="1019" alt="Screenshot 2022-07-28 at 14 11 08" src="https://user-images.githubusercontent.com/4588074/181501794-b2a26c90-0084-4730-99a1-9e27e6042693.png">

Reply:
I've found why the smartcast doesn't work, there is a way to solve by introducing an additional local variable, check this StackOverflow [answer](https://stackoverflow.com/a/69558316).

Reply:
> I've found why the smartcast doesn't work, there is a way to solve by introducing an additional local variable, check this StackOverflow [answer](https://stackoverflow.com/a/69558316).

Great finding, thank you for sharing 😻 . So nice to know the right reason why something we'd be expecting to work isn't in a particular case.

Reply:
Resolved in: https://github.com/wordpress-mobile/WordPress-Android/pull/16934/commits/82b248686d3f17678497fac8933af58915d8e9a4